### ISSUE TITLE
Pyright looks for imports in the first site-packages folder prefixed with "python"

### ISSUE THREAD

**Issue by ipkiss42** [microsoft/pyright#4708]
**Describe the bug**
When looking for the `site-packages` directory; Pyright [looks for the first entry](https://github.com/microsoft/pyright/blob/1cdac3e6641e33ab9218db581a873df7314dd34e/packages/pyright-internal/src/analyzer/pythonPathUtils.ts#L146-L148) whose name starts with `python`, regardless of the Python version.

**To Reproduce**
In a venv created for e.g. Python 3.10, add a `$VENV_PATH/lib/python2.7/site-packages` folder, with the usual contents.

**Expected behavior**
Pyright looks first in the Python version defined in the configuration before looking at other folders.

**VS Code extension or command-line**
Pyright 1.1.296

**Additional context**
I'm not sure how common it is to have multiple `lib/pythonX.Y` folders. If it is too weird or esoteric, feel free to ignore this request.

I reported it because debugging it was non-trivial (some imports were unresolved, even though they could be imported when running the interpreter directly), and I wanted to avoid this debugging to future Pyright users.

In addition, since Pyright already has the python version, I am hoping that prioritizing this version would be easy.

Thanks for your great work on Pyright!

**Comment by erictraut** [comment]
It sounds like you're using the `venv` and `venvPath` mechanism to identify your python environment. Very few pyright users make use of that mechanism and instead configure the default Python environment in their shell prior to launching pyright. Pyright will then execute `python3` and ask it for its import resolution paths (`sys.path`). Is that a viable solution for you?

I didn't realize that it was possible to have multiple lib versions in a single virtual environment. Under what circumstance would that be useful? Isn't a virtual environment always pegged to a single Python interpreter? Do you happen to know if the `lib/pythonX.Y` is documented anywhere, or is it just a convention?

**Comment by ipkiss42** [comment]
> It sounds like you're using the venv and venvPath mechanism to identify your python environment.

That's right.

> Is that a viable solution for you?

I'm not sure. In my case, the Python version and the venv are both known, so having to set up a shell environment sounds like unnecessary configuration. It's also not clear to me how it would work in the IDE, in a multi-root workspace (there is probably a solution though, if it is the most common setup).

> I didn't realize that it was possible to have multiple lib versions in a single virtual environment. Under what circumstance would that be useful?

In my case (at work), some repositories are "packaged" for multiple Python versions, as part of a single "build". The different `pythonX.Y` folders allow avoiding clashes in the build artifacts. Through other processes these build artifacts may be combined, resulting in venv-like folders like this.

> Isn't a virtual environment always pegged to a single Python interpreter?

I don't know for sure. This setup in my work place is the only exception I know of. One might argue that this is not a "real" venv, but in a language with duck typing, if it quacks like a venv... :)

> Do you happen to know if the lib/pythonX.Y is documented anywhere, or is it just a convention?

It is mentioned in the [documentation of the site module](https://docs.python.org/3/library/site.html):

> It starts by constructing up to four directories from a head and a tail part. For the head part, it uses sys.prefix and sys.exec_prefix; empty heads are skipped. For the tail part, it uses the empty string and then lib/site-packages (on Windows) or lib/pythonX.Y/site-packages (on Unix and macOS). For each of the distinct head-tail combinations, it sees if it refers to an existing directory, and if so, adds it to sys.path and also inspects the newly added path for configuration files.

**Comment by erictraut** [comment]
Thanks for the additional information. This will be addressed in the next release.

**Comment by scanny** [comment]
Hi Eric, just some further input, I've removed the `venv` and `venvPath` entries from my `pyrightconfig.json` just now and that fixed things for me. I prefer that actually because my venv details are going to be different from others on the team and this change makes it shareable.

Anyway, the problem I was seeing was in import that couldn't be resolved. The net is that my venv/lib directory looks like this:

```bash
.
..
python3.7
python3.9
site-packages
```

... and **pyright** was looking in `site-packages` and then stopping. Inside that directory was only `pygls`, not sure why, maybe it's version independent or something, but anyway pyright was losing the ability to resolve Python package imports. `pygls` is a language-server library and I installed Ruff recently, so maybe it came from there. That would explain it happening "suddenly".

Anyway, thought you might want to know for when you're in there later or if someone comes up with a sudden problem like this. I diagnosed it by running `pyright --verbose ...` from the command-line, which gave me exactly what I needed to put my finger on the problem, so that's probably good first advice for someone who's seeing a `Could not resolve import` problem arise suddenly :)
